### ISSUE TITLE
"received corrupt message" error on many websites

### ISSUE THREAD

**Issue by Shnatsel** [neonmoe/minreq#48]
On some websites, e.g. http://naturalresourceswales.gov.uk, minreq fails with the following error:

> received corrupt message

Firefox and curl work fine.

90,000+ websites out of the top million from [Feb 3 Tranco list](https://tranco-list.eu/list/3G6L) are affected.

Tested using [this code](https://github.com/Shnatsel/rust-http-clients-smoke-test/blob/f206362f2e81521bbefb84007cdd25242f6db590/minreq-smoke-test/src/main.rs). Test tool output from all affected websites: [minreq-corrupt-message.tar.gz](https://github.com/neonmoe/minreq/files/5992132/minreq-corrupt-message.tar.gz)

**Comment by neonmoe** [comment]
The corrupt message error seems to come from Rustls. Running with `https-bundled-probe` I got the following error, which is a little more verbose:
```rust
IoError(Custom {
    kind: Other,
    error: Failure(Ssl(
        Error {
            code: ErrorCode(1),
            cause: Some(Ssl(ErrorStack([Error {
                code: 336130315,
                library: "SSL routines",
                function: "ssl3_get_record",
                reason: "wrong version number",
                file: "ssl/record/ssl3_record.c",
                line: 331,
            }]))),
        },
        X509VerifyResult {
            code: 0,
            error: "ok",
        },
    )),
})
```
Curious that curl and Firefox avoid this issue.  I tried bumping all the dependencies' versions up, but that didn't fix this either.

**Comment by Shnatsel** [comment]
`ureq` also uses rustls, and it works fine with this website.

**Comment by Shnatsel** [comment]
The exact testing setup I'm using for all clients can be found here: https://github.com/Shnatsel/rust-http-clients-smoke-test

**Comment by neonmoe** [comment]
Based on my testing, this has been fixed by the same change as #49.
